qutip/qip/icm.py

```diff
@@ -0,0 +1,584 @@
+    """
+    gate = cicm.gates[idx]
+    if icm_label(gate.name, gate.arg_label) == "T":
+        t = gate.targets[0]
+        # Move targets and controls of all gates before the T gate
+        for gate in cicm.gates:
+            if gate.arg_label in ["ancilla", "measurement",
+                                  "correction", "teleported"]:
+                continue
+
+            if gate.targets[0] > t:
+                gate.targets[0] += 5
+
+            if gate.name == "CNOT" and gate.controls[0] > t:
+                gate.controls[0] += 5
+
+        cicm.gates[idx] = Gate("a", targets=[t + 1],
+                               arg_label=r"ancilla")
+        cicm.gates.insert(idx + 1, Gate("0", targets=[t + 2],
+                          arg_label=r"ancilla"))
+        cicm.gates.insert(idx + 2, Gate("y", targets=[t + 3],
+                          arg_label=r"ancilla"))
+        cicm.gates.insert(idx + 3, Gate("+", targets=[t + 4],
+                          arg_label=r"ancilla"))
+        cicm.gates.insert(idx + 4, Gate("0", targets=[t + 5],
+                          arg_label=r"ancilla"))
+
+        cicm.gates.insert(idx + 5, Gate("CNOT", targets=[t],
+                          controls=[t+1], arg_label="teleported"))
+        cicm.gates.insert(idx + 6, Gate("CNOT", targets=[t + 2],
+                          controls=[t + 1], arg_label="teleported"))
+        cicm.gates.insert(idx + 7, Gate("CNOT", targets=[t + 1],
+                          controls=[t + 3], arg_label="teleported"))
+        cicm.gates.insert(idx + 8, Gate("CNOT", targets=[t + 2],
+                          controls=[t + 4], arg_label="teleported"))
+        cicm.gates.insert(idx + 9, Gate("CNOT", targets=[t + 5],
+                          controls=[t + 3], arg_label="teleported"))
+        cicm.gates.insert(idx + 10, Gate("CNOT", targets=[t + 5],
+                          controls=[t + 4], arg_label="teleported"))
+        cicm.gates.insert(idx + 11, Gate("z", targets=[t],
+                          arg_label="measurement"))
+        cicm.gates.insert(idx + 12, Gate("z/x", targets=[t + 1],
+                          controls=[t], arg_label="correction"))
+        cicm.gates.insert(idx + 13, Gate("x/z", targets=[t + 2],
+                          controls=[t+1], arg_label="correction"))
+        cicm.gates.insert(idx + 14, Gate("x/z", targets=[t + 3],
+                          controls=[t+2], arg_label="correction"))
+        cicm.gates.insert(idx + 15, Gate("z/x", targets=[t + 4],
+                          controls=[t+3], arg_label="correction"))
+
+        # Move targets and controls of all gates after the T gate
+
+        for gate in cicm.gates[idx+16:]:
+            if gate.arg_label in ["ancilla", "measurement",
+                                  "correction", "teleported"]:
+                continue
+            if gate.targets[0] == t:
+                gate.targets[0] += 5
+
+        cicm.N += 5
+    return (cicm)
+
+
+def visualise(icm_circuit):
```

Comment:
This function is not used anywhere.
Should this be included in the class? Is it needed?